src/daemon/daemon.cpp

```diff
@@ -1742,25 +1757,26 @@ try // clang-format on
             timestamp->set_nanos(snapshot->get_creation_timestamp().time().msec() * 1'000'000);
         };
 
-        if (const auto& it = instance_snapshots_map.find(name);
-            it == instance_snapshots_map.end() || it->second.empty())
-        {
-            for (const auto& snapshot : vm.view_snapshots())
-                get_snapshot_info(snapshot);
-        }
-        else
+        const auto& it = instance_snapshots_map.find(name);
+        const auto& [pick, all] = it == instance_snapshots_map.end() ? SnapshotPick{{}, true} : it->second;
+
+        try
         {
-            for (const auto& snapshot : it->second)
+            if (all)
```

Comment:
I think the if/else clauses need to be switched around, no?

Reply:
Hmm, maybe I am overlooking something, but this seems correct to me. `all` is supposed to be true if we need to fetch all the snapshots for the instance called `name`. But before we do so, we need to verify that any arguments referring to individual snapshots of the same instance are valid, so that we complain on `multipass info --snapshot-overview x x.missing`.

Reply:
ah, I see! Alright, I think that's ok.